src/javascripts/Views/editColumnsModal.js

```diff
@@ -39,15 +39,16 @@ ngapp.controller('editColumnsModalController', function($scope, xelibService, co
 
     $scope.close = function() {
         $scope.allColumns.forEach(function(column) {
-            if (column.custom) columnsService.buildDataFunction(column);
+            if (column.custom) $scope.columnsService.buildDataFunction(column);
         });
         $scope.$root.$broadcast('rebuildColumns');
-        columnsService.saveColumns();
+        $scope.columnsService.saveColumns();
```

Comment:
passing a service through modalOptions is kind of gross.  We definitely need to use a single generic `columnsService`